pyomo/common/deprecation.py

```diff
@@ -247,6 +254,15 @@ def deprecated(msg=None, logger=None, version=None, remove_in=None):
         remove_in (str): the version in which the decorated object will be
             removed from the code.
 
+    Example
+    -------
+    >>> from pyomo.common.deprecation import deprecated
+    >>> @deprecated('The sample_function has been deprecated.', version='1.2.3')
+        class sample_function(x):
+            return x
```

Comment:
This isn't valid python.  Does Sphinx not test this code?

Proposed suggestion:
```python
>>> from pyomo.common.deprecation import deprecated
>>> @deprecated(version='1.2.3')
... def sample_function(x):
...     return 2*x
>>> sample_function(5)
WARNING: DEPRECATED: This function (sample_function) has been deprecated and
    may be removed in a future release.  (deprecated in 1.2.3) ...
10
```
```